oauth2/src/main/java/com/linecorp/armeria/common/auth/oauth2/TokenRevocation.java

```diff
@@ -40,26 +41,44 @@ public static TokenRevocationBuilder builder(WebClient revocationEndpoint, Strin
         return new TokenRevocationBuilder(revocationEndpoint, revocationEndpointPath);
     }
 
-    private final TokenRevocationRequest request;
+    private final OAuth2Endpoint<Boolean> revocationEndpoint;
+    @Nullable
+    private final Supplier<ClientAuthentication> clientAuthenticationSupplier;
 
-    TokenRevocation(WebClient revocationEndpoint, String revocationEndpointPath,
-                    @Nullable ClientAuthorization clientAuthorization) {
-        request = new TokenRevocationRequest(revocationEndpoint, revocationEndpointPath, clientAuthorization);
+    TokenRevocation(OAuth2Endpoint<Boolean> revocationEndpoint,
+                    @Nullable Supplier<ClientAuthentication> clientAuthenticationSupplier) {
+        this.revocationEndpoint = revocationEndpoint;
+        this.clientAuthenticationSupplier = clientAuthenticationSupplier;
     }
 
     /**
      * Implements Token Revocation request/response flow,
      * as per <a href="https://datatracker.ietf.org/doc/html/rfc7009#section-2">[RFC7009], Section 2</a>.
      */
     public CompletableFuture<Boolean> revokeRefreshToken(String refreshToken) {
-        return request.make(refreshToken, REFRESH_TOKEN);
+        final ClientAuthentication clientAuthentication = clientAuthentication();
+        final TokenOperationRequest tokenOperationRequest =
+                TokenOperationRequest.of(clientAuthentication, refreshToken, REFRESH_TOKEN);
+        return revocationEndpoint.execute(tokenOperationRequest);
     }
 
     /**
      * Implements Token Revocation request/response flow,
      * as per <a href="https://datatracker.ietf.org/doc/html/rfc7009#section-2">[RFC7009], Section 2</a>.
      */
     public CompletableFuture<Boolean> revokeAccessToken(String accessToken) {
-        return request.make(accessToken, ACCESS_TOKEN);
+        final ClientAuthentication clientAuthentication = clientAuthentication();
+        final DefaultTokenOperationRequest tokenOperationRequest =
+                new DefaultTokenOperationRequest(clientAuthentication, accessToken, REFRESH_TOKEN);
```

Comment:
So that the nullibility of `accessToken` is verified

```suggestion
                TokenOperationRequest.of(clientAuthentication, accessToken, REFRESH_TOKEN);
```

Reply:
Good point. A factory method should be preferred.